src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -637,8 +637,9 @@ COptTasks::OptimizeTask(void *ptr)
 		CRefCount::SafeRelease(trace_flags);
 		CRefCount::SafeRelease(plan_dxl);
 		CMDCache::Shutdown();
-
-		IErrorContext *errctxt = CTask::Self()->GetErrCtxt();
+		CTask *task = CTask::Self();
+		GPOS_ASSERT(NULL != task);
```

Comment:
GPOS_ASSERT in the GPOS_CATCH_EX block? Maybe make a condition: if CTask::Self() returns NULL, then don’t use errctxt?
Also, I think that you shouldn’t have mixed such different changes in one commit!

Reply:
Fix moved to https://github.com/arenadata/gpdb/pull/856